In [1]:
import pandas as pd

In [2]:
import numpy as np

In [3]:
data=pd.DataFrame({'productA':[25,30,28,32,26,29,27,33,31,28,30,29],'ProductB':[40,35,38,41,37,36,39,42,38,37,40,36]})

In [4]:
data

productA  ProductB
0         25        40
1         30        35
2         28        38
3         32        41
4         26        37
5         29        36
6         27        39
7         33        42
8         31        38
9         28        37
10        30        40
11        29        36

## Measures of Central Tendency :

#### 1. Calculate the mean sales for Product A and Product B. Compare the average monthly sales for the two products and discuss any observed trends.

In [5]:
A = data.iloc[:,0]

In [6]:
A

0     25
1     30
2     28
3     32
4     26
5     29
6     27
7     33
8     31
9     28
10    30
11    29
Name: productA, dtype: int64

In [7]:
B = data.iloc[:,1]

In [8]:
B

0     40
1     35
2     38
3     41
4     37
5     36
6     39
7     42
8     38
9     37
10    40
11    36
Name: ProductB, dtype: int64

In [9]:
mean_A = np.mean(data.iloc[:,0])

In [10]:
mean_A

29.0

In [11]:
mean_B = np.mean(data.iloc[:,1])

In [12]:
mean_B

38.25

In [13]:
data.mean()

productA    29.00
ProductB    38.25
dtype: float64

* Product A has an average monthly sales of 29.0. 
* product B has an average monthly sales of 38.25. 
* This means that product B sells more each month compared to product A, making it the better performer in terms of sales.

#### 2. Find the median sales for each product and discuss how it reflects the central tendency of the sales distribution.

In [19]:
median_A = np.median(data.iloc[:,0])

In [20]:
median_A

29.0

In [21]:
median_B = np.median(data.iloc[:,1])

In [22]:
median_B

38.0

In [23]:
data.median()

productA    29.0
ProductB    38.0
dtype: float64

median of product B is middle value it indicates that below median sales are low and above medium sales are high median shows a clear middle point in the sales data for both products

#### 3. Identify the mode of the sales data for both Product A and Product B. Discuss the significance of the mode in the context of these products.

In [28]:
data.mode()

productA  ProductB
0      28.0        36
1      29.0        37
2      30.0        38
3       NaN        40

In [29]:
print(data.mode())

   productA  ProductB
0      28.0        36
1      29.0        37
2      30.0        38
3       NaN        40


* The mode is the sales figure that happens most often in the dataset. 
* For Product A, the mode is 30, which means 30 is the most frequent sales value. 
* For Product B, the mode is 40, indicating that 40 is the most common sales figure. 
* When the mode is close to the mean and median, it suggests that the sales distribution is relatively symmetrical. 

#### 4. Calculate the weighted mean of the sales data, assuming each month has an equal weight. Discuss any implications of using weighted mean in this context.

In [30]:
A_sales = np.array([25,30,28,32,26,29,27,33,31,28,30,29])

In [31]:
B_sales = np.array([40,35,38,41,37,36,39,42,38,37,40,36])

In [36]:
weights = np.ones(len(A_sales))/len(A_sales)

In [37]:
weighted_mean = np.average(A_sales, weights = weights)

In [38]:
weighted_mean # for products A

28.999999999999996

In [39]:
weights1 = np.ones(len(B_sales))/len(B_sales)

In [41]:
weighted_mean1 = np.average(B_sales, weights = weights)

In [42]:
weighted_mean1 # for products B

38.25

## Measures of Dispersion:

#### 1. Compute the range of monthly sales for both Product A and Product B. Discuss how the range provides insights into the variability of sales.

In [59]:
A = data.iloc[:,0]

In [60]:
A

0     25
1     30
2     28
3     32
4     26
5     29
6     27
7     33
8     31
9     28
10    30
11    29
Name: productA, dtype: int64

In [61]:
B = data.iloc[:,1]

In [62]:
B

0     40
1     35
2     38
3     41
4     37
5     36
6     39
7     42
8     38
9     37
10    40
11    36
Name: ProductB, dtype: int64

In [65]:
range_A = np.ptp(A) # peek to peek = calculates max - min.

0     25
1     30
2     28
3     32
4     26
5     29
6     27
7     33
8     31
9     28
10    30
11    29
Name: productA, dtype: int64

In [66]:
print('Range :', range_A)

Range : 8


In [67]:
range_B = np.ptp(B)

In [68]:
print('Range :', range_B)

Range : 7


#### 2. Calculate the variance of the sales data for both products. Discuss what the variance reveals about the spread of sales for each product.

In [69]:
variance_A = np.var(A)

In [70]:
variance_A

5.166666666666667

In [71]:
variance_B = np.var(B)

In [72]:
variance_B

4.354166666666667

#### 3.  Determine the standard deviation of the sales for Product A and Product B. Discuss the significance of standard deviation in understanding the variability in sales.

In [73]:
std_A = np.std(A)

In [74]:
std_A

2.273030282830976

In [75]:
std_B = np.std(B)

In [76]:
std_B 

2.0866640042581523

## Relative Measures of Dispersion: 

In [77]:
A = data.iloc[:,0]

In [78]:
A

0     25
1     30
2     28
3     32
4     26
5     29
6     27
7     33
8     31
9     28
10    30
11    29
Name: productA, dtype: int64

In [79]:
B = data.iloc[:,1]

In [80]:
B

0     40
1     35
2     38
3     41
4     37
5     36
6     39
7     42
8     38
9     37
10    40
11    36
Name: ProductB, dtype: int64

#### 1. Calculate the coefficient of variation for the sales data of Product A. Discuss how this relative measure of dispersion helps in comparing the variability of products with different average sales.

In [98]:
mean_A = np.mean(A)

In [99]:
mean_A

29.0

In [100]:
std_A = np.std(A)

In [101]:
std_A

2.273030282830976

In [102]:
cv_A = std_A/mean_A * 100

In [103]:
cv_A

7.838035458037848

In [104]:
print('coefficient of varaince for product A : ', cv_A)

coefficient of varaince for product A :  7.838035458037848


#### 2.  Compute the relative range for Product B. Discuss how the relative range accounts for the scale of the sales data.

In [105]:
max_B = max(B)
min_B = min(B)

In [107]:
max_B,min_B 

(42, 35)

In [106]:
range_B = max_B - min_B

In [108]:
range_B

7

In [109]:
mean_B = np.mean(B)

In [110]:
mean_B

38.25

In [111]:
relative_range = (range_B/mean_B)*100

In [112]:
relative_range

18.30065359477124

#### 3. Determine the relative standard deviation for Product A. Interpret the result and discuss its implications for understanding the variability relative to the mean.

In [113]:
mean_A = np.mean(A)

In [114]:
mean_A 

29.0

In [115]:
std_A = np.std(A)

In [116]:
std_A

2.273030282830976

In [117]:
relative_standard_deviation = (std_A/mean_A)*100

In [118]:
relative_standard_deviation

7.838035458037848